plugins/outputs/opensearch/opensearch.go

```diff
@@ -331,6 +339,82 @@ func createBulkIndexer(osInst *Opensearch, pipelineName string) (opensearchutil.
 	return opensearchutil.NewBulkIndexer(bulkIndexerConfig)
 }
 
+func (o *Opensearch) GetIndexName(rawIndexName string, metric telegraf.Metric) (string, error) {
+	var tagVal = func(key string) interface{} {
+		if val, ok := metric.Tags()[key]; ok {
+			return val
+		}
+		return o.DefaultTagValue
+	}
+	var fieldVal = func(key string) interface{} {
+		if val, ok := metric.Fields()[key]; ok {
+			return val
+		}
+		return ""
+	}
+	var nameVal = func() string{
+		return metric.Name()
+	}
+
+	var funcs = template.FuncMap{
+		"Name": nameVal,
+		"Tag": tagVal,
+		"Field": fieldVal,
+	}
+	
+	var indexTmpl, err = template.New("index").Funcs(funcs).Parse(rawIndexName)
+	if err != nil {
+		return "", fmt.Errorf("parsing index name failed: %w", err)
+	}
+
+	var buf bytes.Buffer
+	err = indexTmpl.Execute(&buf, struct{Time time.Time}{metric.Time().UTC()})
+	if err != nil {
+		return "", fmt.Errorf("creating index name failed: %w", err)
+	}
+	var indexName = buf.String()
+	if strings.Contains(indexName, "{{") {
+		return "", fmt.Errorf("failed to evaluate valid indexname: %s", indexName)
+	}
+	o.Log.Debugf("indexName- %s", indexName)
+	
+
+	return indexName, nil
+}
+
+func (o *Opensearch) getPipelineName(metric telegraf.Metric) (string, error) {
+	if o.UsePipeline == "" || !strings.Contains(o.UsePipeline, "{{") {
+		return o.UsePipeline, nil
+	}
+	var tagVal = func(key string) interface{} {
+		if val, ok := metric.Tags()[key]; ok {
+			return val
+		}
+		return o.DefaultTagValue
+	}
+	
+	var pipelineTmpl, err = template.New("pipeline").Funcs(template.FuncMap{"Tag": tagVal}).Parse(o.UsePipeline)
+	if err != nil {
+		return "", fmt.Errorf("parsing pipeline name failed: %w", err)
+	}
```

Comment:
Same here, initialising the template parser should occur in `Init()`.

Reply:
please refer [above](https://github.com/influxdata/telegraf/pull/11958#discussion_r1324206922), made a similar call with dummy metric, which will validate the pipeline.

Reply:
done